runtime/metricsview/executor.go

```diff
@@ -127,10 +127,40 @@ func (e *Executor) ValidateQuery(qry *Query) error {
 	panic("not implemented")
 }
 
-// Watermark returns the current watermark of the metrics view.
-// If the watermark resolves to null, it defaults to the current time.
-func (e *Executor) Watermark(ctx context.Context) (time.Time, error) {
-	return e.loadWatermark(ctx, nil)
+// Timestamps queries min, max and watermark for the metrics view
+func (e *Executor) Timestamps(ctx context.Context, executionTime *time.Time) (time.Time, time.Time, time.Time, error) {
+	if !e.min.IsZero() {
+		return e.min, e.max, e.watermark, nil
+	}
+
+	var err error
+	switch e.olap.Dialect() {
+	case drivers.DialectDuckDB:
+		e.min, e.max, e.watermark, err = e.resolveDuckDB(ctx)
+	case drivers.DialectDruid:
+		e.min, e.max, e.watermark, err = e.resolveDruid(ctx)
+	case drivers.DialectClickHouse:
+		e.min, e.max, e.watermark, err = e.resolveClickHouseAndPinot(ctx)
+	case drivers.DialectPinot:
+		e.min, e.max, e.watermark, err = e.resolveClickHouseAndPinot(ctx)
+	default:
+		return time.Time{}, time.Time{}, time.Time{}, fmt.Errorf("not available for dialect '%s'", e.olap.Dialect())
+	}
+	if err != nil {
+		return time.Time{}, time.Time{}, time.Time{}, err
+	}
+	if executionTime != nil {
+		e.watermark = *executionTime
+	}
```

Comment:
A tricky problem here is that in many cases (like most alerts/reports), only the watermark is used, so this override here actually means the timestamps query was redundant.

Any thoughts on a simple way to optimize that? Otherwise we can go with this for now and think about optimizing this later.